client/lib/domains/assembler.js

```diff
@@ -23,7 +23,7 @@ function createDomainObjects( dataTransferObject ) {
 		return {
 			autoRenewalDate: domain.auto_renewal_date,
 			expirationDate: domain.expiry,
-			expirationMoment: domain.expiry && i18n.moment( domain.expiry, 'MMMM D, YYYY', 'en' ).locale( false ),
+			expirationMoment: domain.expiry ? i18n.moment( domain.expiry ) : null,
```

Comment:
I don't think we need both `expirationDate` and `expirationMoment` - especially since the first one is actually never used in code :laughing:
Looking at the backend code, it's not possible for `domain.expiry` to be null here, because we are only fetching domains that are mapped to the site, so either the domain has been bought with us (and, naturally, has a domain expiration date) or has been mapped (and the subscription's expiration date becomes `domain.expiry`). So we shouldn't be needing the check for `domain.expiry` here (IMHO - I might have missed something).
And even if we have this check it doesn't do us any good, because we don't check if `expirationMoment` is null when we actually try to invoke its methods :grin: